src/meili/config.cc

```diff
@@ -0,0 +1,113 @@
+#include "meili/config.h"
+#include "midgard/logging.h"
+
+#define CHECK_POSITIVE_PARAM(value, name)                                                            \
```

Comment:
Macro is a great idea, but I think we can generalize this. My thoughts: 
- Add a `macro.h` header ... somewhere. I'm not sure what folder to put this in. 
- Generalize `CHECK_POSITIVE_PARAM` to a more general conditional check, like a `CHECK_THROW(<condition>, <message>)` macro. You could probably define something like:

```
#define CHECK_ARGUMENT_THROW(condition, exc_info)  \
  do {                                                                                           \
    if (!(condition)) {                                                                    \
      throw std::invalid_argument(exc_info);                             \
    }                                                                                              \
  } while (0)
```

I probably butchered the syntax here, but you get the idea.

Reply:
done. the main idea here was to avoid copying an exception message